cephfs: add e2e and documentation for cephFS static PVC (#1059)
ceph/ceph-csi: docs/static-pvc.md

**Note** deleting PV and PVC doesnot deleted the backend rbd image, user need to
manually delete the rbd image if required

## CephFS static PVC

CephFS volumes created manually can be mounted and unmounted to an app, below step

> shows how to create a cephFS volume, static PV and static PVC

Reviewer: @Madhu-1 I dont think we need to change it, `CephFS` is the correct usage.

> shows how to create a cephFS volume, static PV and static PVC

Reviewer 2: Its not cephfs volume it's cephfs subvolume isn't it

> shows how to create a cephFS volume, static PV and static PVC

Reviewer: @Madhu-1 @agarwal-mudit yeah, subvolume is correct, but I was mainly sharing the thought on the usage of "CephFS" and capitalization in it instead of `cephFS" as in this PR which looks wrong to me. @nixpanic @batrick what's the correct usage?  Is it `CephFS` or `cephFS`" ,that could help us to have a consistent mention of it in the repo.

> shows how to create a cephFS volume, static PV and static PVC

Reviewer 3: "CephFS" is correct

> ### Create cephFS sub-volume

Reviewer: cephFS/CephFS

If you already have a cephFS sub-volume created and contains some data
which you want to access by the application pod you can skip this step.

Lets create a new cephFS volume in ceph cluster which we are going to use for
static PVC, before that we need to create the subvolumegroup

```bash
[$]ceph fs subvolumegroup create <volume-name> <subvolumegroup-name>
```

```bash

> [$]ceph fs subvolume create <volume-name> <subvolume-name> <subvolumegroup-name> --size=1024

Reviewer 2: subvolume  size is in byes, not MB

> [$]ceph fs subvolume create <volume-name> <subvolume-name> <subvolumegroup-name> --size=1024

Reviewer: byes/bytes :)

```

**Note** volume here refers to the filesystem.

### Create cephFS static PV

> To create the cephFS PV you need to know the `volume root path`, and `clusterID`,

Reviewer 2: `volume rootpath`

here is the command to get the root path in ceph cluster

> ```bash

Reviewer 2: add example

[$]ceph fs subvolume getpath <volume-name> <subvolume-name> <subvolumegroup-name>
```

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: cephfs-static-pv
spec:
  accessModes:
  - ReadWriteMany
  capacity:
    storage: 1Gi
  csi:
    driver: cephfs.csi.ceph.com
    nodeStageSecretRef:
      # node stage secret name

>       name: csi-cephfs-secret

Reviewer 2: please add more info about this secret, this secret need userID and userKey

      # node stage secret namespace where above secret is created
      namespace: default
    volumeAttributes:
      # Required options from storageclass parameters need to be added in volumeAttributes
      "clusterID": "ba68226a-672f-4ba5-97bc-22840318b2ec"
      "staticVolume": "true"

>       "rootPath: /volumes/csi/testVol

Reviewer 2: ```suggestion
      "rootPath": "/volumes/csi/testVol"
```

**Note** deleting PV and PVC doesnot deleted the backend rbd image, user need to
manually delete the rbd image if required

## CephFS static PVC

> CephFS sub-volume created manually can be mounted and unmounted to an app,

Reviewer 2: sub-volume is a single word.

Reviewer 2: this need to be handled in all places

> below steps show how to create a CephFS volume, static PV and static PVC.

Reviewer 2: ```suggestion
below steps show how to create a CephFS subvolume, static PV and static PVC.
```

### Create CephFS sub-volume

If you already have a CephFS sub-volume created and contains some data
which you want to access by the application pod you can skip this step.

Lets create a new CephFS sub-volume of size 1 GiB in ceph cluster which
we are going to use for static PVC, before that we need to create

> the subvolumegroup. **myfs** is the file system name(volume name) inside

Reviewer 2: ```suggestion
the subvolumegroup. **myfs** is the filesystem name(volume name) inside
which subvolume should be created.
```

which subvolume should be created.

> ```bash

Reviewer 2: if there is no output from command use console instead of bash. need same changes in other places also

[$]ceph fs subvolumegroup create myfs testGroup
```

```bash
[$]ceph fs subvolume create myfs testSubVolume testGroup --size=1073741824
```

**Note:** volume here refers to the filesystem.

### Create CephFS static PV

To create the CephFS PV you need to know the `volume rootpath`, and `clusterID`,
here is the command to get the root path in ceph cluster

```bash
[$]ceph fs subvolume getpath myfs testSubVolume testGroup
/volumes/testGroup/testSubVolume
```

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: cephfs-static-pv
spec:
  accessModes:
  - ReadWriteMany
  capacity:
    storage: 1Gi
  csi:
    driver: cephfs.csi.ceph.com
    nodeStageSecretRef:
      # node stage secret name
      name: csi-cephfs-secret
      # node stage secret namespace where above secret is created
      namespace: default
    volumeAttributes:
      # Required options from storageclass parameters need to be added in volumeAttributes
      "clusterID": "ba68226a-672f-4ba5-97bc-22840318b2ec"
      "fsName": "myfs"
      "staticVolume": "true"

>       "rootPath": /volumes/csi/testVol

Reviewer 2: will this be testVol or testSubVolume as per above example?

    # volumeHandle can be anything, need not to be same
    # as PV name or volume name. keeping same for brevity
    volumeHandle: cephfs-static-pv
  persistentVolumeReclaimPolicy: Retain
  volumeMode: Filesystem
```

### Node stage secret ref in CephFS PV

> The rook created secrets will not have the keys userID and userKey.

Reviewer 2: we don't mention about rook in cephcsi, cephcsi is independent of the rook.
